### DISCUSSION TITLE
custom ids are returned empy array on create

### DISCUSSION THREAD

**Discussion by f-i-t-s-u-m** [feathersjs/feathers#3049] [Q&A] [answered]
## Description
I am using FeathersJS with a Knex MySQL database, and I am trying to implement UUID as the primary key for my table. However, when I create a new record, the response from the create method is an empty array ([]) instead of the newly created data object.

## Steps to Reproduce
Create a FeathersJS service that uses Knex MySQL as the database.
Set the primary key for the table to be a UUID.
In the create method, return the newly created data object.
Test the service by sending a POST request to the endpoint, and observe the response.

## Expected Result
The response from the create method should be the newly created data object, including the newly generated UUID.

## Actual Result
The response from the create method is an empty array ([]), and the newly created data object is not returned.

## Additional Information
I have tried modifying the create method to explicitly retrieve the newly created record from the database and return it in the response, but I encountered the error "The values in the where clause must not be an object or an array". I also tried casting the id value to a string before using it in the where clause, but I encountered the error "ER_NO_TABLES_USED: No tables used".

## Environment
FeathersJS version: 5.0.0-pre.37
Knex MySQL version: 5.0.0-pre.37

## Questions
Can you provide a solution for this issue, or suggest a workaround?
If there is a bug in the FeathersJS code, can you suggest a fix?
Thank you for your help!

**Comment by daffl** [comment, marked as answer]
This has been addressed and then reverted in https://github.com/feathersjs/feathers/pull/2934 because it breaks functionality in other databases. I found the most reliable solution is to generate the UUID (or any other id format) in the `data` resolver before it is created in the database.

We could add the id returning behaviour as an option to avoid the whack-a-mole we've been doing with the various fixes around this.

**Reply by f-i-t-s-u-m** [reply]
okay thank you @daffl my issue has been fixed. 
I have been using feathers since release of 4. keep up the good work.
